src/System.Threading.Channels/tests/Stress.cs

```diff
@@ -0,0 +1,423 @@
+            bool shouldReadAllWrittenValues = boundedOptions == null || boundedOptions.FullMode == BoundedChannelFullMode.Wait;
+
+            List<Task> taskList = new List<Task>();
+
+            int readerTasksCount;
+            int writerTasksCount;
+
+            if (options.SingleReader)
+            {
+                readerTasksCount = 1;
+                writerTasksCount = options.SingleWriter ? 1 : MaxTaskCounts - 1;
+            }
+            else if (options.SingleWriter)
+            {
+                writerTasksCount = 1;
+                readerTasksCount = MaxTaskCounts - 1;
+            }
+            else
+            {
+                readerTasksCount = MaxTaskCounts / 2;
+                writerTasksCount = MaxTaskCounts - readerTasksCount;
+            }
+
+            int readCount = 0;
+            int wroteCount = 0;
+
+            for (int i=0; i < readerTasksCount; i++)
+            {
+                taskList.Add(new Task(() => {
+                    while (true)
+                    {
+                        try
+                        {
+                            if (!reader.TryRead(out int value))
+                            {
+                                value = reader.ReadAsync().GetAwaiter().GetResult();
+                            }
+                            Interlocked.Increment(ref readCount);
+                        }
+                        catch (ChannelClosedException)
+                        {
+                            break;
+                        }
+                    }
+                }));
+            }
+
+            int numberToWriteToQueue = -1;
+            for (int i=0; i < writerTasksCount; i++)
+            {
+                taskList.Add(new Task(() => {
+                    while (true)
+                    {
+                        int incrementedValue = Interlocked.Increment(ref numberToWriteToQueue);
+                        if (incrementedValue >= MAX_NUMBER_TO_WRITE_TO_CHANNEL)
+                        {
+                            // we have reached the max we can write.
+                            if (incrementedValue == MAX_NUMBER_TO_WRITE_TO_CHANNEL)
+                            {
+                                while (wroteCount < MAX_NUMBER_TO_WRITE_TO_CHANNEL)
+                                {
+                                    // we have some tasks is currently writing and not finished yet. wait them to finish.
+                                    Task.Delay(100);
+                                }
+
+                                writer.Complete();
+                            }
+                            break;
+                        }
+
+                        if (!writer.TryWrite(incrementedValue))
+                        {
+                            writer.WriteAsync(incrementedValue).GetAwaiter().GetResult();
+                        }
+                        Interlocked.Increment(ref wroteCount);
```

Comment:
There is a simpler way to coordinate the writers.  You just need to have the last one call writer.Complete, e.g.
```C#
int remainingWriters = writerTasksCount;
...
taskList.Add(Task.Run(async delegate
{
    int i = 0;
    while (i < MaxNumberToWriteToChannel)
    {
        await writer.WriteAsync(i);
        i = Interlocked.Increment(ref numberToWriteToQueue);
    }
    if (Interlocked.Decrement(ref remainingWriters) == 0) writer.Complete();
});
```

Reply:
ok. will do that